diff --git build-system/Make/Make.py build-system/Make/Make.py
--- build-system/Make/Make.py
+++ build-system/Make/Make.py
@@ -248,6 +248,9 @@ class BazelCommandLine:
                 '--features=swift.split_derived_files_generation',
             ]
 
+        if self.additional_args is not None:
+            combined_arguments += self.additional_args
+
         return combined_arguments
 
     def invoke_build(self):
@@ -618,6 +621,9 @@ def build(bazel, arguments):
     bazel_command_line.set_profile_swift(arguments.profileSwift)
 
     bazel_command_line.set_split_swiftmodules(arguments.enableParallelSwiftmoduleGeneration)
 
+    if arguments.bazelArguments is not None:
+        bazel_command_line.add_additional_args(shlex.split(arguments.bazelArguments))
+
     bazel_command_line.invoke_build()
